### PR TITLE
#146 - Set default log level for non-prod environment to DEBUG

### PR DESCRIPTION
**Pull request by MackHalliday**
# Description

Issue #146 

- Update logic in `logger.py` `init_app` method to set log levels to `INFO` if production, else `DEBUG` is not.
- Update unit testing

## Type of change

Please check the relevant option(s).

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation changes only
- [x] Internal

## How Has This Been Tested?

How has this been tested? (e.g. Unit tests, Tested locally, Tested as a GitHub action, Depoyed to dev, etc)  
Please provide relevant information and / or links to demonstrate the functionality or fix. (e.g. screenshots, link to deployment, regression test results, etc)

- Ran locally with Postman. I toggled the environment from `production` to `development` and confirmed the logging level changed according to the environment. 
- Deployed `notification-api` to DEV with the dependency for `notification-utils` referencing this feature branch (146-update_debug_log_level ). Confirmed that DEV logs now showed DEBUG logs.  [See the branch I used for testing here. ](https://github.com/department-of-veterans-affairs/notification-api/pull/1843)

image
image

## Checklist

- [x] I have assigned myself to this PR
- [x] PR has an appropriate title: #9999 - What the thing does
- [x] PR has a detailed description, including links to specific documentation
- [x] I have added the [appropriate labels](https://github.com/department-of-veterans-affairs/notification-api/blob/master/.github/release.yaml) to the PR.
- [x] I did not remove any parts of the template, such as checkboxes even if they are not used
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [N/A] I have made corresponding changes to any documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have ensured the latest master is merged into my branch and all checks are green prior to review
- [x] New and existing unit tests pass locally with my changes
- [N/A] Any dependent changes have been merged and published in downstream modules
- [x] The ticket is now moved into the DEV test column

### CODE REVIEW

File: scripts/run_tests.sh [hunk 1]
```diff
@@ -22,8 +22,8 @@ function display_result {
   fi
 }
 
-flake8 --config .flake8 .
-display_result $? 1 "Code style check"
+# flake8 --config .flake8 .
+# display_result $? 1 "Code style check"
```
Comment (MackHalliday) [inline scripts/run_tests.sh:26]: Temporarily comment out

File: notifications_utils/logging.py [hunk 2]
```diff
@@ -80,11 +80,16 @@ def after_request(response):
 
     ensure_log_path_exists(app.config['NOTIFY_LOG_PATH'])
     the_handler = get_handler(app)
+
+    # Add to see log level while running local 
     loglevel = logging.getLevelName(app.config['NOTIFY_LOG_LEVEL'])
+    print("This is the log level", loglevel)
```
Comment (MackHalliday) [inline notifications_utils/logging.py:86]: This does not seem to show up in logs when running locally in Docker

File: notifications_utils/logging.py [hunk 3]
```diff
@@ -1,12 +1,14 @@
 import logging
```
Comment (MackHalliday) [inline notifications_utils/logging.py:1]: pep8 sort

File: notifications_utils/logging.py [hunk 4]
```diff
@@ -87,7 +90,17 @@ def after_request(response):
         the_logger.setLevel(loglevel)
     logging.getLogger('boto3').setLevel(logging.WARNING)
     logging.getLogger('s3transfer').setLevel(logging.WARNING)
-    app.logger.info("Logging configured")
+
+    app.logger.info("Logging configured. The log level has been set to %s", app.logger.level)
+
+
+def set_log_level(app):
```
Comment (MackHalliday) [inline notifications_utils/logging.py:97]: logic to set log level

File: tests/test_logging.py [hunk 5]
```diff
@@ -1,10 +1,38 @@
...
 import logging as builtin_logging
+import os
 import uuid
+
+import pytest
 from notifications_utils import logging
 from pythonjsonlogger.jsonlogger import JsonFormatter
 
 
+class App:
```
Comment (MackHalliday) [inline tests/test_logging.py:11]: Create shared test resources

File: tests/test_logging.py [hunk 6]
```diff
@@ -159,3 +171,17 @@ class App:
     assert message_dict["message"] == "Hello, Cornelius.  Line 42."
     assert message_dict["pathname"] == "the_path"
     assert message_dict["lineno"] == 1999
+
+
+def test_it_set_log_level_production(app, reset_environment):
```
Comment (MackHalliday) [inline tests/test_logging.py:176]: Tests for setting log level

File: notifications_utils/logging.py [hunk 7]
```diff
@@ -87,7 +90,17 @@ def after_request(response):
...
+    # For production environment, set log level to INFO
+    # For all other environment, set log level to DEBUG
```
Comment (kalbfled) [inline notifications_utils/logging.py:99]:
```suggestion
    """
    For production environment, set log level to INFO.
    For all other environment, set log level to DEBUG.
    """
```

**Review by kalbfled — CHANGES_REQUESTED** [review]

**Review by kalbfled — APPROVED** [review]

**Review by k-macmillan — APPROVED** [review]

**Pull request merged**